### PR TITLE
5.0 update of intro to MVC

### PR DESCRIPTION
**Pull request by Rick-Anderson**
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/start-mvc?view=aspnetcore-5.0&branch=pr-en-us-20428)

### CODE REVIEW

File: aspnetcore/tutorials/first-mvc-app/adding-model.md
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app/adding-model.md:74]: As soon as this merges to master, fix the preceding lines for the right version.

**Comment by Rick-Anderson** [comment]
@wadepickett  please test this for show stopper bugs only. Keep in mind we'll fix  the following as soon as it merges:

image

**Comment by wadepickett** [comment]
@Rick-Anderson:

Adding-controller.md:     Section:  Add a controller:   VS tab:

Instruction:
"In Solution Explorer, right-click Controllers > Add > Controller"

However, the screenshot highlights "New Item..." menu instead.  Screenshot needs to be fixed.
image

**Comment by Rick-Anderson** [comment]
@wadepickett  none of those are show stoppers. Fix them when you go through this series  later in the week. We need to get the 5.0 version out.

**Comment by wadepickett** [comment]
Thanks.  I respectfully disagree.  Those are incorrect instructions that can't be followed.   I'm am checking through the rest of it of course, times 3 IDE's.  I am ignoring all the issues in your doc that I am finding that are not completely wrong or blocking, or just don't work.  I am just sticking to the high level scope of blockers.

**Comment by Rick-Anderson** [comment]
I've only updated the first 3, don't review beyond that.

> hose are incorrect instructions that can't be followed. 

Folks are following them now without reporting that. You'll have a chance to fix them soon.

File: aspnetcore/tutorials/first-mvc-app/adding-view.md
Comment (wadepickett) [reply aspnetcore/tutorials/first-mvc-app/adding-view.md:91]: For the New File instruction for Mac:
Change Select "MVC View Page" to "Razor View".  Folks new to this are going to have no idea it is "Razor View" now until they look it up somewhere else.  They will assume they are missing templates until they find out on on the forums that this is just wrong.

**Review by wadepickett — CHANGES_REQUESTED** [review]

**Comment by wadepickett** [comment]
@Rick-Anderson, I stuck just blockers, ignored any other problems,  for all 3 IDE's.  Let me know when you want me to look beyond tutorial 3.  I had also checked "adding-model.md" since you hand submitted it.

**Comment by Rick-Anderson** [comment]
> In order for scaffolding to run we need to make sure the audience has the following package added first:
Microsoft.VisualStudio.Web.CodeGenerators.Mvc" Version="5.0.0-rc.2.20516.1"

Can you add that as a suggest comment and I'll accept?

**Comment by Rick-Anderson** [comment]
> Since the moniker range above this 5.0 specific range also includes 5.0, the markup we ask the audience to follow is repeated twice, showing both the 3.x and the 5.0 version.

Good one, check that my last commit fixed that.

**Comment by Rick-Anderson** [comment]
> However, the screenshot highlights "New Item..." menu instead. Screenshot needs to be fixed, or the instruction. The screenshot following this one assumes "New Item.." was selected.

See https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/tutorials/first-mvc-app/adding-controller/_static

@wadepickett  the images in aspnetcore\tutorials\first-mvc-app\adding-controller\_static are correct, both of them. When you go through this you can figure out why they're not being used.

**Comment by wadepickett** [comment]
@Rick-Anderson, yes, fix the text so it is correct. and not mismatched and wrong instruction.

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson, yes, fix the text so it is correct. and not mismatched and wrong instruction.

Did you look at the images? The text is correct, the images are correct. I don't see why the wrong images are rendering.

> Let me know when you want me to look beyond tutorial 3. 

You mean 4 right?

**Comment by wadepickett** [comment]
@Rick-Anderson "I don't see why the wrong images are rendering."   
If you are not seeing the updated images in the browser, but you know they are in the build.  Dump the browser cache.  I have noticed Edge holding on to old images since the last Edge update even after refresh.

**Comment by wadepickett** [comment]
@Rick-Anderson "Let me know when you want me to look beyond tutorial 3.

You mean 4 right?"

Yes, by "beyond tutorial 3", I mean any tutorials you want me to review that come sequentially after 3.

**Comment by Rick-Anderson** [comment]
@wadepickett  you reviewed **Part 4, add a model to an ASP.NET Core MVC app** which is beyond 3

File: aspnetcore/tutorials/first-mvc-app/adding-controller.md
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app/adding-controller.md:34]: @wadepickett this is the correct image, can you figure out why it's not rendering. It's a year old.

File: aspnetcore/tutorials/first-mvc-app/adding-controller.md
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app/adding-controller.md:38]: @wadepickett this is also the correct image, but not what is being rendered.

**Comment by wadepickett** [comment]
I will create a new issue to track any broken instruction called out here that will not be fixed.   Given we are short on time, I will approve, but that is assuming you fix the missing package issue. Don't use the full preview version number of course since this is pretty much going out with the update.

**Review by wadepickett — APPROVED** [review]
Given we are short on time, I will move any unfixed blockers to a separate issue to track to fix.  Thanks.

**Comment by Rick-Anderson** [comment]
> In order for scaffolding to run we need to make sure the audience has the following package added first:
Microsoft.VisualStudio.Web.CodeGenerators.Mvc" Version="5.0.0-rc.2.20516.1"

According to @deepchoudhery  all the scaffolding issues are fixed in RTM 5.0

VS doesn't require adding packages before scaffolding.

I didn't hit that issue when I went through this, I did however, hit [these two](https://github.com/dotnet/Scaffolding/issues/1450)

File: aspnetcore/tutorials/first-mvc-app/adding-model.md
Comment (Rick-Anderson) [reply aspnetcore/tutorials/first-mvc-app/adding-model.md:159]: > In order for scaffolding to run we need to make sure the audience has the following package added first:
> Microsoft.VisualStudio.Web.CodeGenerators.Mvc" Version="5.0.0-rc.2.20516.1"
> 
> Otherwise they will get this error:
> image

The images were correct. I sent you links to show that. For some reason it kept rendering the incorrect images. I made copies of each year old image and reference the copy image, now it's rendering correctly.  Not sure how it could cache an image more than a year old.

**Comment by wadepickett** [comment]
Scaffolding.  That is great it is fixed in RTM 5.0.  It seemed weird to required the additional package after the install.

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson:
> 
> Adding-controller.md: Section: Add a controller: VS tab:
> 
> Instruction:
> "In Solution Explorer, right-click Controllers > Add > Controller"
> 
> However, the screenshot highlights "New Item..." menu instead. Screenshot needs to be fixed, or the instruction. The screenshot following this one assumes "New Item.." was selected.
> image
> 
> Also, "MVC Controller Class" has been renamed in the latest to just **MVC Controller- Empty** hasn't it? The instruction and the screenshot for "Add New Item" looks like it needs to be updated, otherwise folks are looking for something that no longer exists:
> 
> This is what the updated screen would contain:
> image

The images were correct. I sent you links to show that. For some reason it kept rendering the incorrect images. I made copies of each year old image and reference the copy image, now it's rendering correctly.  Not sure how it could cache an image more than a year old.

File: aspnetcore/tutorials/first-mvc-app/working-with-sql.md
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app/working-with-sql.md:6]: @wadepickett this is ready

Comment (wadepickett) [reply aspnetcore/tutorials/first-mvc-app/working-with-sql.md:6]: ...on this.  VS was OK, ran into an issue with VSCode/VS Mac but going back to the beginning to to try again to see if I just missed something.  Will have it soon.

File: aspnetcore/tutorials/first-mvc-app/adding-model.md
Comment (wadepickett) [inline aspnetcore/tutorials/first-mvc-app/adding-model.md:72]: This calls out the new targetFramework "net5.0" as I would expect.   However, later when I run dotnet aspnet-codegenerator I get an error indicating "Ensure that restore has run and that you have included 'net50' in the TargetFrameworks for your project. "   If I change the TargetFramework to "net50", it no longer gives me that error, but then instead I get a scaffolding failed error: Scaffolding failed.
"Please add Microsoft.VisualStudio.Web.CodeGeneration.Design package to the project as a NuGet package reference."    However I do have the latest added as a reference.   I assume you tested and it worked for you.  If so then disregard the issue I am running into.

Comment (deepchoudhery) [reply aspnetcore/tutorials/first-mvc-app/adding-model.md:72]: Hmm yeah that was an issue with older preview versions of the 5.0 sdk. Definitely have not run into something like that in a while. Are you testing using the rtm sdk?

Comment (wadepickett) [reply aspnetcore/tutorials/first-mvc-app/adding-model.md:72]: Thanks, I'll go back and check.  I had switched machines, which I thought were both up to date.

Comment (wadepickett) [reply aspnetcore/tutorials/first-mvc-app/adding-model.md:72]: @deepchoudhery, my apologies for the mistake.  My 2nd machine I had switched to was on 5.0.0-rc.2.20475.5.  I know better, sorry about that.

Comment (Rick-Anderson) [reply aspnetcore/tutorials/first-mvc-app/adding-model.md:72]: @deepchoudhery  where can we get the RTM 5.0 SDK?

Comment (Rick-Anderson) [reply aspnetcore/tutorials/first-mvc-app/adding-model.md:72]: > @deepchoudhery where can we get the RTM 5.0 SDK?

Got it, never mind.

Comment (wadepickett) [reply aspnetcore/tutorials/first-mvc-app/adding-model.md:72]: @Rick-Anderson, when you tested for VS Code, were you able to get past scaffolding fine?  I have been unable to with this PR plus the very latest .NET.  I may just have something messed up here that I'm not spotting.  If your test passed great.

**Pull request merged**
